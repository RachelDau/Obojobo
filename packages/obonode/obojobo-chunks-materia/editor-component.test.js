import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import IFrame from './editor-component'

import ModalUtil from 'obojobo-document-engine/src/scripts/common/util/modal-util'
jest.mock('obojobo-document-engine/src/scripts/common/util/modal-util')

describe('IFrame Editor Node', () => {
	beforeEach(() => {
		jest.restoreAllMocks()
		jest.resetAllMocks()
	})
	test('IFrame component', () => {
		const component = renderer.create(
			<IFrame
				node={{
					data: {
						get: () => ({
							width: 200,
							height: 200,
							controls: '',
							border: false,
							initialZoom: 1,
							src: 'mockSrc',
							title: 'mockTitle'
						})
					}
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('IFrame renders with no size correctly', () => {
		const component = renderer.create(
			<IFrame
				node={{
					data: {
						get: () => ({
							controls: '',
							border: false,
							initialZoom: 1,
							src: 'mockSrc'
						})
					}
				}}
			/>
		)
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('IFrame component edits properties', () => {
		const change = {
			removeNodeByKey: jest.fn()
		}

		const component = mount(
			<IFrame
				node={{
					data: {
						get: () => ({
							controls: '',
							border: false,
							initialZoom: 1,
							src: ''
						})
					}
				}}
				editor={{
					value: { change: () => change },
					onChange: jest.fn()
				}}
			/>
		)

		component
			.find('button')
			.at(0)
			.simulate('click')

		expect(ModalUtil.show).toHaveBeenCalled()

		component.unmount()
	})

	test('changeProperties sets the nodes content', () => {
		const editor = {
			setNodeByKey: jest.fn()
		}

		const component = mount(
			<IFrame
				node={{
					data: {
						get: () => ({
							controls: '',
							border: false,
							initialZoom: 1
						})
					}
				}}
				editor={editor}
			/>
		)

		component.instance().changeProperties({ mockProperties: 'mock value' })

		expect(editor.setNodeByKey).toHaveBeenCalled()
	})
})
