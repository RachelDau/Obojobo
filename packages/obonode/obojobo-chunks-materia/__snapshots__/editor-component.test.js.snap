// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IFrame Editor Node IFrame component 1`] = `
<div
  className="obojobo-draft--chunks--iframe viewer pad is-previewing  is-not-bordered is-not-showing  is-controls-enabled  is-not-missing-src is-not-scaled-up"
>
  <div
    className="editor-container"
    style={
      Object {
        "height": "200px",
      }
    }
  >
    <div
      className="iframe-toolbar"
    >
      <span
        aria-hidden={true}
        className="title"
      >
        mockTitle
      </span>
      <div
        className="obojobo-draft--components--button is-not-dangerous align-center properties-button"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          IFrame Properties
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`IFrame Editor Node IFrame renders with no size correctly 1`] = `
<div
  className="obojobo-draft--chunks--iframe viewer pad is-previewing  is-not-bordered is-not-showing  is-controls-enabled  is-not-missing-src is-not-scaled-up"
>
  <div
    className="editor-container"
    style={
      Object {
        "height": "500px",
      }
    }
  >
    <div
      className="iframe-toolbar"
    >
      <span
        aria-hidden={true}
        className="title"
      >
        mockSrc
      </span>
      <div
        className="obojobo-draft--components--button is-not-dangerous align-center properties-button"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          IFrame Properties
        </button>
      </div>
    </div>
  </div>
</div>
`;
